lib/chat_models/chat_anthropic.ex

```diff
@@ -201,11 +205,21 @@ defmodule LangChain.ChatModels.ChatAnthropic do
     # Anthropic sets the `system` message on the request body, not as part of the messages list.
     |> Utils.conditionally_add_to_map(:system, system_text)
     |> Utils.conditionally_add_to_map(:tools, get_tools_for_api(tools))
+    |> Utils.conditionally_add_to_map(:tool_choice, set_tool_choice(anthropic))
     |> Utils.conditionally_add_to_map(:max_tokens, anthropic.max_tokens)
     |> Utils.conditionally_add_to_map(:top_p, anthropic.top_p)
     |> Utils.conditionally_add_to_map(:top_k, anthropic.top_k)
   end
 
+  defp set_tool_choice(%ChatAnthropic{tool_choice: %{"type" => "tool", "name" => name}=_tool_choice}) when is_binary(name) and byte_size(name) > 0,
+    do: %{"type" => "tool", "name" => name}
+
+  defp set_tool_choice(%ChatAnthropic{tool_choice: %{"type" => type}=_tool_choice}) when type != nil and is_binary(type),
+    do: %{"type" => type}
+
+  defp set_tool_choice(%ChatAnthropic{}), do: nil
```

Comment:
Maybe `get_tool_choice` would be a better name?

Reply:
Agree with you that it's a better name. Modified it accordingly.

Also added the tests for live calls.